vterm-module.c

```diff
@@ -216,7 +216,22 @@ static bool is_eol(Term *term, int end_col, int row, int col) {
   }
   return 1;
 }
-
+static int is_end_of_prompt(Term *term, int end_col, int row, int col) {
+  LineInfo *info = get_lineinfo(term, row);
+  if (info == NULL) {
+    return 0;
+  }
+  if (info->prompt_col < 0) {
```

Comment:
It seems this LineInfo is defined in vterm-module.h and never actually set, or used (until now). So I think the supporting code for tracking the prompt is missing,

Reply:
Oh, I am wrong! It is set, but it is expecting OCS callback to help it!

Reply:
see https://github.com/akermu/emacs-libvterm#directory-tracking

Directory tracking and Prompt tracking requires some configuration, as the shell has to be
instructed to share the relevant information with Emacs.  
For `zsh`, put this at the end of your `.zshrc`:

```zsh
vterm_prompt_end() {
    vterm_printf "51;A$(whoami)@$(hostname):$(pwd)";
}
setopt PROMPT_SUBST
PROMPT=$PROMPT'%{$(vterm_prompt_end)%}'
```